lib/Conversion/ImportVerilog/Statement.cpp

```diff
@@ -0,0 +1,149 @@
+//===- Statement.cpp - Slang statement conversion--------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "ImportVerilogInternals.h"
+#include "slang/ast/ASTVisitor.h"
+#include "slang/ast/Symbol.h"
+#include "slang/ast/symbols/CompilationUnitSymbols.h"
+#include "slang/ast/symbols/InstanceSymbols.h"
+#include "slang/ast/symbols/VariableSymbols.h"
+#include "slang/ast/types/AllTypes.h"
+#include "slang/ast/types/Type.h"
+#include "slang/syntax/SyntaxVisitor.h"
+
+using namespace circt;
+using namespace ImportVerilog;
+
+void Context::convertExpression(const slang::ast::Expression *expression) {
+  auto loc = convertLocation(expression->sourceRange.start());
+  switch (expression->kind) {
+  case slang::ast::ExpressionKind::IntegerLiteral:
+    assert(0 && "TODO");
+    break;
+  case slang::ast::ExpressionKind::NamedValue:
+    // I think that needs to call another function, like visitExpr(...), which
+    // handles details about expressions.
+    assert(0 && "TODO");
+    break;
+  case slang::ast::ExpressionKind::UnaryOp:
+    assert(0 && "TODO");
+    break;
```

Comment:
Can you change these to something like
```
return mlir::emitError(loc, "unsupported expression: unary operator");
```

The reason is that asserts get disabled in release builds, so you might end up never really getting any feedback that something went wrong. Reporting an error diagnostic and returning with an error makes the failure properly pop up.

Reply:
I got it. Thank you very much for your advice. I will correct them.